### PR TITLE
Fix 9918

### LINKED ISSUE #9918: [doc] transformers.PreTrainedTokenizer.encode() doesn't get resolved to its doc
There are a few 

> See: transformers.PreTrainedTokenizer.encode()

in the docstrings, but they don't resolve to anything in the online docs, since the `encode` method is in `PreTrainedTokenizerBase`
as it can be seen: https://huggingface.co/transformers/internal/tokenization_utils.html#transformers.tokenization_utils_base.PreTrainedTokenizerBase.encode

Should sphinx be able to resolve inheritance and still point to the right doc, or must the docs be modified to say:

> See: transformers.PreTrainedTokenizerBase.encode()

instead?

There are 117 of these.

Example:

https://huggingface.co/transformers/model_doc/t5.html#transformers.T5Model.forward

> Indices can be obtained using T5Tokenizer. See transformers.PreTrainedTokenizer.encode() and transformers.PreTrainedTokenizer.__call__() for detail.

the `encode` method doesn't get a link.

-----------------------

Also it seems that all modules have this

> `What are input IDs? <../glossary.html#input-ids>`__

but not where `input_ids` are documented, instead after the mask - shouldn't that line be 2 records up?

snapshot_18

@sgugger

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR addresses the problem shown in #9918 by:
- adding the documentation of the `encode` method to the `PreTrainedTokenizer` and `PreTrainedTokenizerFast` (it is in all their subclasses already)
- adding the "What are input IDs" link where missing in some models docstrings.

In passing, I uncovered a failure of the doc styling script on DPR, so this PR also fixes that.

Fixes #9918

### CODE REVIEW

File: src/transformers/models/dpr/modeling_tf_dpr.py [hunk 1]
```diff
@@ -486,22 +486,26 @@ def serving(self, inputs):
 
             (a) For sequence pairs (for a pair title+text for example):
 
-                ``tokens: [CLS] is this jack ##son ##ville ? [SEP] no it is not . [SEP]``
+            ::
```
Comment (sgugger) [inline src/transformers/models/dpr/modeling_tf_dpr.py:489]: That docstring had been changed to work around the doc-styling failure. This makes it the same as the PyTorch one,

File: utils/style_doc.py [hunk 2]
```diff
@@ -135,6 +135,14 @@ def init_in_block(self, text):
         """
         return SpecialBlock.NOT_SPECIAL
 
+    def end_of_special_style(self, line):
+        """
+        Sets back the `in_block` attribute to `NOT_SPECIAL`.
+
+        Useful for some docstrings where we may have to go back to `ARG_LIST` instead.
+        """
+        self.in_block = SpecialBlock.NOT_SPECIAL
```
Comment (sgugger) [inline utils/style_doc.py:144]: Basically the problem was that when a code block was nested in the argument list, we were returning to a not special style instead of returning to an arg list style. This new method overriden in the subclass `DocStyler` ensures we return to the proper type.

**Review by LysandreJik — APPROVED** [review]
Great, thanks for updating all of these! LGTM!
